README.md

```diff
@@ -17,11 +32,449 @@ Xcode 16 or later.
+```
+
+The second argument to `rooms.get` is a `RoomOptions` argument, which tells the Chat SDK what features you would like your room to use and
+how they should be configured.
+
+You can also use `RoomOptions.allFeaturesEnabled` to enable all room features with the default configuration.
+
+For example, you can set the timeout between keystrokes for typing events as part of the room options. Sensible defaults for each of the
+features are provided for your convenience:
+
+- A typing timeout (time of inactivity before typing stops) of 5 seconds.
+- Entry and subscription to presence.
+
+Here’s an example demonstrating how to specify a custom typing timeout of 3 seconds:
+
+```swift
+let room = try await chatClient.rooms.get(roomID: "basketball-stream",
+                                          options: .init(typing: TypingOptions(timeout: 3.0)))
+```
+
+In order to use the same room but with different options, you must first `release` the room before requesting an instance with the changed
+options (see below for more information on releasing rooms).
+
+Note that:
+
+- If a `release` call is currently in progress for the room (see below), then a call to `get` will wait for that to complete before completing
+  itself.
+- If a `get` call is currently in progress for the room and `release` is called, the `get` call will fail.
+
+### Attaching to a room
+
+To start receiving events on a room, it must first be attached. This can be done using the `attach` method:
+
+```swift
+try await room.attach()
+```
+
+### Detaching from a room
+
+To stop receiving events on a room, it must be detached, which can be achieved by using the `detach` method:
+
+```swift
+room.detach()
+```
+
+Note: This does not remove any event listeners you have registered and they will begin to receive events again in the
+event that the room is re-attached.
+
+### Releasing a room
+
+Depending on your application, you may have multiple rooms that come and go over time (e.g. if you are running 1:1 support chat). When you
+are completely finished with a room, you may `release` it which allows the underlying resources to be collected:
+
+```swift
+_ = try await rooms.release(roomID: "basketball-stream")
+```
+
+Once `release` is called, the room will become unusable and you will need to get a new instance using `rooms.get`.
+
+> [!NOTE]
+> Releasing a room may be optional for many applications. If release is not called, the server will automatically tidy up
+> connections and other resources associated with the room after a period of time.
+
+### Monitoring room status
+
+Monitoring the status of the room is key to a number of common chat features. For example, you might want to display a warning when the room
+has become detached.
+
+### Current status of a room
+
+To get the current status (and error), you can use the `status` property like this:
+
+```swift
+switch try await room.status {
```

Comment:
Why is there a `try`?

Reply:
https://github.com/ably/ably-chat-swift/compare/6472e96566e4bb8a82ad6b336fbc7718a0a14557..189d5d7aac1d8ce0019baab657b9473d8268182f